lib/auto-languageclient.ts

```diff
@@ -279,7 +279,34 @@ export default class AutoLanguageClient {
     await this._serverManager.stopAllServers();
   }
 
-  protected spawnChildNode(args: string[], options: cp.SpawnOptions = {}): cp.ChildProcess {
+  /**
+   * Spawn a general language server.
+   * Use this inside the `startServerProcess` override if the language server is a general executable.
+   * Also see the `spawnChildNode` method.
+   * If the name is provided as the first argument, it checks `bin/platform-arch/exeName` by default, and if doesn't exists uses the exe on PATH.
+   * For example on Windows x64, by passing `serve-d`, `bin/win32-x64/exeName.exe` is spawned by default.
+   * @param exe the `name` or `path` of the executable
+   * @param args args passed to spawn the exe. Defaults to `[]`.
+   * @param options: child process spawn options. Defaults to `{}`.
+   * @param rootPath the path of the folder of the exe file. Defaults to `join("bin", `${process.platform}-${process.arch}`)`.
+   * @param exeExtention the extention of the exe file. Defaults to `process.platform === "win32" ? ".exe" : ""`
+   */
+  protected spawn(
```

Comment:
Should this be called `spawnChildExe` to compliment `spawnChildNode`?

Reply:
I think `spawnChildNode` comes from the fact that Atom's Node process (a real child) is used for spawning. But `spawn` is general (the spawned exe is independent), and using `Child` can be misleading.